packages/http/src/validator/validators/body.ts

```diff
@@ -1,82 +1,141 @@
 import { IPrismDiagnostic } from '@stoplight/prism-core';
-import { IMediaTypeContent } from '@stoplight/types';
-import { get } from 'lodash';
-import { parse as deepParseUrlEncoded } from 'qs';
+import { DiagnosticSeverity, Dictionary, IHttpEncoding, IMediaTypeContent } from '@stoplight/types';
+import * as Array from 'fp-ts/lib/Array';
+import * as Either from 'fp-ts/lib/Either';
+import * as Option from 'fp-ts/lib/Option';
+import { pipe } from 'fp-ts/lib/pipeable';
+import { get, partial } from 'lodash';
 import * as typeIs from 'type-is';
-import { body } from '../deserializers';
 
-import { DiagnosticSeverity, Dictionary } from '@stoplight/types';
+import { fromEquals } from 'fp-ts/lib/Eq';
 import { JSONSchema } from '../../types';
-import { validateAgainstSchema } from '../validators/utils';
+import { body } from '../deserializers';
 import { IHttpValidator } from './types';
+import { validateAgainstSchema } from './utils';
 
 export class HttpBodyValidator implements IHttpValidator<any, IMediaTypeContent> {
-  constructor(private _prefix: string) {}
+  constructor(private prefix: string) {}
 
   public validate(target: any, specs: IMediaTypeContent[], mediaType?: string): IPrismDiagnostic[] {
-    const { _prefix: prefix } = this;
-    const content = this.getContent(specs, mediaType);
-
+    const content = getContentByMediaTypeOrFirst(specs, mediaType);
     const schema = get(content, 'schema');
 
-    if (!schema) {
-      return [];
-    }
-
-    if (mediaType && typeIs.is(mediaType, 'application/x-www-form-urlencoded')) {
-      const shallowParsedTarget = shallowParseUrlEncoded(target);
-      target = deepParseUrlEncoded(target);
-
-      const encodings = get(content, 'encodings', []);
-      for (const encoding of encodings) {
-        const allowReserved = get(encoding, 'allowReserved', false);
-        const property = encoding.property;
-        const value = shallowParsedTarget[property];
-
-        if (!allowReserved && typeof value === 'string' && value.match(/[\/?#\[\]@!$&'()*+,;=]/)) {
-          return [
-            {
-              path: [prefix, property],
-              message: 'Reserved characters used in request body',
-              severity: DiagnosticSeverity.Error,
-            },
-          ];
-        }
+    return pipe(
+      Either.fromNullable([])(schema),
+      Either.map(() => target),
```

Comment:
`target` is already available from the function parameter and does not need any change or "fp-isation", so we can just use that directly.